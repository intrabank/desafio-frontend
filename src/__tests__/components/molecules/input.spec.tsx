import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import Input from '@/components/molecules/input';
import { render, screen } from '@testing-library/react';

describe('Input', () => {
	it('should render input', () => {
		const tree = renderer
			.create(<Input label='Teste' htmlFor='teste' />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('should render input with error', () => {
		render(
			<Input label='Teste' placeholder='teste' htmlFor='teste' state='error' />
		);

		const input = screen.getByPlaceholderText('teste');
		const label = screen.getByText('Teste');

		expect(input.getAttribute('aria-invalid')).toBe('true');
		expect(label.innerHTML).toBe('Teste<span> *</span>');
	});

	it('should render input with error', () => {
		render(
			<Input label='Teste' placeholder='teste' htmlFor='teste' state='active' />
		);

		const input = screen.getByPlaceholderText('teste');
		const label = screen.getByText('Teste');

		expect(input.getAttribute('aria-invalid')).toBe('false');
		expect(label.innerHTML).toBe('Teste');
	});

	it('should render input with toggle password type', () => {
		render(
			<Input
				label='Teste'
				placeholder='teste'
				htmlFor='teste'
				type='password'
				state='active'
			/>
		);

		const input = screen.getByPlaceholderText('teste');
		const label = screen.getByText('Teste');

		expect(input.getAttribute('aria-invalid')).toBe('false');
		expect(label.innerHTML).toBe('Teste');
	});

	it('should render error message', () => {
		render(
			<Input
				label='Teste'
				placeholder='teste'
				htmlFor='teste'
				type='password'
				error='Error message'
				state='error'
			/>
		);

		const errorMessage = screen.getByText('Error message');

		expect(errorMessage).toBeInTheDocument();
	});
});
