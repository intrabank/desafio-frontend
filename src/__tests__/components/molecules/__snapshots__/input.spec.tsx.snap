// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render input 1`] = `
<div
  className="c-UFKVs"
>
  <label
    className="c-bUsXAP c-bUsXAP-fXZMCZ-state-idle"
    htmlFor="teste"
    onMouseDown={[Function]}
  >
    Teste
  </label>
  <input
    aria-invalid={false}
    aria-placeholder="Teste"
    className="c-gpMnof c-gpMnof-gFpoIP-state-idle"
    type="text"
  />
</div>
`;
